### PR TITLE
Build DMG files with `appdmg`

### PR DESCRIPTION
**Pull request by fluiddot**
Related to https://github.com/Automattic/studio/pull/135.

## Proposed Changes

- Use `appdmg` library to create dmg files.

## Testing Instructions

- Observe CI jobs succeed.
- Observe `Build for Mac` job generates dmg files.
- Download and run dmg files to confirm they work as expected.

## Pre-merge Checklist

- [x] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

**Comment by fluiddot** [comment]
I'm encountering the same issue spotted [here](https://github.com/Automattic/studio/pull/135#issuecomment-2112913759). The problem is related to a native module used in the package `fs-xattr`. Seems npm builds native modules for the current architecture only. As a workaround, we could try to enforce the needed architecture (`x86` or `arm64`) based on the matrix.

File: .buildkite/commands/install-node-dependencies.sh [hunk 1]
```diff
@@ -42,6 +42,9 @@ npm ci \
   --maxsockets "$MAX_SOCKETS" \
   "$@"
 
+# Rebuid specific native modules to avoid failures due to using different architectures
+npm rebuild fs-xattr --verbose --cpu universal
```
Comment (fluiddot) [inline .buildkite/commands/install-node-dependencies.sh]: This workaround works locally but I couldn't make it work on CI 😞 . In theory, if we rebuild the native module using the architecture `x64` or `universal`, there shouldn't be any failures when generating the DMG file for `x64` or `universal` versions.

**Comment by mokagio** [comment]
Just a screenshot to show the correct output in the `arm64` build:

image

That is encouraging 🙂

Unfortunately, I don't have much to add in terms of addressing the other issues...

**Comment by fluiddot** [comment]
Merging this PR as I've tested DMG files in https://github.com/Automattic/studio/pull/321.

**Pull request merged**